src-ocaml/baijiu_blake2b.ml

```diff
@@ -362,38 +356,43 @@ module Make (B : Baijiu_buffer.S)
       ; buf
       ; h = Array.make 8 0L
       ; t = Array.make 2 0L
-      ; f = Array.make 2 0L }
-    in
+      ; f = Array.make 2 0L } in
 
     let param_bytes = param_to_bytes
         { default_param with digest_length = outlen
                            ; key_length = len } in
 
     for i = 0 to 7
-    do ctx.h.(i) <- Int64.(iv.(i) lxor (B.le64_to_cpu param_bytes (i * 8))) done;
+    do ctx.h.(i) <- Int64.(iv.(i) lxor (By.le64_to_cpu param_bytes (i * 8))) done;
 
     if len > 0
     then begin
-      let block = B.create 128 in
+      let block = By.create 128 in
 
-      B.fill block 0 128 '\x00';
-      B.blit key off block 0 len;
+      By.fill block 0 128 '\x00';
+      blit key off block 0 len;
 
-      feed ctx block 0 128;
+      unsafe_feed_bytes ctx block 0 128;
     end;
 
     ctx
 
-  let get ctx =
-    let res = B.create 64 in
+  let with_outlen_and_bytes_key outlen key off len =
+    with_outlen_and_key ~blit:By.blit outlen key off len
+
+  let with_outlen_and_bigstring_key outlen key off len =
+    with_outlen_and_key ~blit:By.blit_from_bigstring outlen key off len
+
+  let unsafe_get ctx =
+    let res = By.create 64 in
```

Comment:
I think it would be nicer if `let res = By.create 64 in` was moved down to line 394 where it gets used, since the part that immediately follows it is not using it, and `res` still contains data leaked from the heap at this point.